### DISCUSSION TITLE
Is migrating from NX to PNPM viable in my situation?

### DISCUSSION THREAD

**Discussion by jkavalec** [pnpm/pnpm#8393] [Q&A]
Hi,

I have a been happily using NX monorepo on several projects, I want to ditch it for one project, and looking for options how to do it and whether PNPM could help me with it.

- Project was developed as monorepo so it is mostly libraries
- I want to release some of these libraries as open-source
- I don't want to release applications build on these as open-source
- I want to have them on Github as separate project
- I want to be able to keep the structure, so that these projects have share root folder when I develop locally, i.e. when I make change to library, all the libraries that depend on this will be using the latest version immediately

I was for instance experimenting with yarn's link where I linked local libs and this worked, however this would need some transformation before pushing to github, to change dependencies to actual NPM dependencies, which is doable but I am first looking for simpler solution.

Does anyone have experience with similar migration? And could PNPM be set up in a way that would fit the above requirements?

**Comment by zkochan** [comment]
I would recommend to check out [bit.cloud](https://bit.cloud/). I think it satisfies all your needs. All your components (or packages) are hosted separately but you can combine and develop them locally in any combination using the Bit workspace. Updates are also handled by Bit (but that is a paid feature). 

You probably can achieve the same with pnpm but you'll need to build some infrastructure of your own arounds it. We don't have built in mechanics for such complex scenarios.

**Comment by jkavalec** [comment]
Thank you for suggestion.

I was looking working with this a bit more and it seems that this will be achieved with a little bit of work on my side.
Most important and unpleasant part that is having this workspace manager outside project and managing/linking contained dependencies is something pnpm does really well, so I am now sure ditching NX in my monorepo and going full on pnpm will be perfectly viable.
